taker/src/routes.rs

```diff
@@ -139,8 +140,18 @@ pub async fn post_order_request(
     taker: &State<Taker>,
     _auth: Authenticated,
 ) -> Result<(), HttpApiProblem> {
+    // TODO: remove me. This is a temporary fix to ensure no one tries to use a different leverage
+    // for now
+    if cfd_order_request.leverage != Leverage::TWO {
+        return Err(HttpApiProblem::new(StatusCode::BAD_REQUEST)
+            .title("Invalid leverage selected. For time being, please go with leverage 2."));
+    }
     taker
-        .take_offer(cfd_order_request.order_id, cfd_order_request.quantity)
+        .take_offer(
+            cfd_order_request.order_id,
+            cfd_order_request.quantity,
+            cfd_order_request.leverage,
+        )
```

Comment:
❓ Do we really need this?

Reply:
Why not? Because it goes towards #1739 ?

Reply:
Sorry, I meant the `if` statement specifically. We can't verify on the maker side that the requested leverage is acceptable?